packages/ic-vetkd-utils/src/lib.rs

```diff
@@ -37,6 +35,34 @@ pub struct TransportSecretKey {
     secret_key: Scalar,
 }
 
+/// Derive a symmetric key using HKDF-SHA256
+pub fn derive_symmetric_key(input: &[u8], domain_sep: &str, len: usize) -> Vec<u8> {
+    let hk = hkdf::Hkdf::<sha2::Sha256>::new(None, input);
+    let mut okm = vec![0u8; len];
+    hk.expand(domain_sep.as_bytes(), &mut okm)
+        .expect("Unsupported output length for HKDF");
+    okm
+}
+
+fn hash_to_scalar(input: &[u8], domain_sep: &str) -> ic_bls12_381::Scalar {
+    use ic_bls12_381::hash_to_curve::HashToField;
+
+    let mut s = [ic_bls12_381::Scalar::zero()];
+    <ic_bls12_381::Scalar as HashToField>::hash_to_field::<ExpandMsgXmd<sha2::Sha256>>(
+        input,
+        domain_sep.as_bytes(),
+        &mut s,
+    );
+    s[0]
+}
+
+fn prefix_with_len(bytes: &[u8]) -> Vec<u8> {
+    let mut out = Vec::with_capacity(bytes.len() + 8);
+    out.extend_from_slice(&(bytes.len() as u64).to_be_bytes());
+    out.extend_from_slice(bytes);
+    out
+}
+
 #[cfg_attr(feature = "js", wasm_bindgen)]
 impl TransportSecretKey {
```

Comment:
As discussed in the team meeting: maybe add a way to serialize and deserialize the TransportSecretKey, to be on par with the Typescript library.

Reply:
Added serialization and deserialization to `TransportSecretKey`